src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java

```diff
@@ -0,0 +1,5 @@
+package io.github.cottonmc.energy.api;
+
+public interface EnergyAttributeProvider {
+    EnergyAttribute getEnergy();
```

Comment:
This, sadly, won't work. An Item is, for all intents and purposes, static; you never store any info on the item itself. I think it'd be best if this was passed a compound tag, so that it could be used on both items and entities simultaneously. Also, once it's merged, I'll write some javadocs for it.

Reply:
Like add a to/fromTag methods and a simple impl for them?

Reply:
Yeah, instead of `EnergyAttribute getEnergy()` do `EnergyAttribute fromTag(CompoundTag tag)` and `void toTag(EnergyAttribute attribute, CompoundTag tag)`.

Reply:
I think I'll leave the getEnergy() as an easy-to-use one for entities and just give it a simple default impl using the fromTag()

Reply:
Should it have default impls?

Reply:
I don't think so.

Reply:
Don't merge yet, one more change